src/client/Client.cc

```diff
@@ -7956,6 +7956,12 @@ int Client::readlinkat(int dirfd, const char *relpath, char *buf, loff_t size, c
     return r;
   }
 
+  if (!strcmp(relpath, "")) {
```

Comment:
Since the open(2) docs speaks about fchownat(2), fstatat(2), linkat(2) too (for the special case with symlink fd and empty pathname), I guess those func in client code should too have this change?

Reply:
Probably, and the corresponding APIs would be `ceph_chownat` and `ceph_statxat`. _libcephfs_ doesn't seem to have an implementation for [linkat(2)](https://www.man7.org/linux/man-pages/man2/linkat.2.html) to create hard links relative to file descriptor.

I can come up with follow up changes(PRs) to handle the case with `AT_EMPTY_PATH` flag for `ceph_chownat` and `ceph_statxat` later.

Reply:
Good catch. I think we need those implemented for completeness sake 👍

Reply:
> Good catch. I think we need those implemented for completeness sake 👍

https://github.com/ceph/ceph/pull/59936 has been created.